transcoder/src/thumbnails.go

```diff
@@ -128,26 +157,73 @@ func extractThumbnail(path string, sha string) error {
 		)
 	}
 
-	// Cleanup old versions of thumbnails
-	files, err := filepath.Glob(getThumbGlob(sha))
-	if err == nil {
-		for _, f := range files {
-			// ignore errors
-			os.Remove(f)
-		}
+	// Track which items existed before creating the new ones.
+	// It is impotant not to delete the items before writing the new ones,
+	// or requests for the old items will fail in the time between when the
+	// old items are deleted and the new ones are created.
+	oldItems, err := s.storage.ListItemsWithPrefix(ctx, getThumbPrefix(sha))
+	if err != nil {
+		return fmt.Errorf("failed to get old thumbnail items: %w", err)
 	}
```

Comment:
files could be removed upfront, this function holds a lock that prevent others from reading the file while it's processing.

(anyway this path only happens when we change the `ThumbsVersion` & want to delete all thumbnails generated with a prior version of the code)

Reply:
Is this desirable in all cases? For example:
* If saving the thumbnail files fail
* If image encoding fails
* If the process is restarted while in this section

If it is desirable to remove the contents in these cases, then I can simplify this as suggested. WDYT?

Reply:
Yes i think this is desirable, anyways we don't change thumbs code often so this case shouldn't be encountered frequently.

let's do with the simplest possible